src/plugins/dashboard/public/dashboard_app/top_nav/editor_menu.tsx

```diff
@@ -261,141 +219,122 @@ export const EditorMenu = ({
+      description,
     };
   };
 
   const getVisTypeAliasMenuItem = (
+    onClickCb: () => void,
     visTypeAlias: VisTypeAlias
-  ): EuiContextMenuPanelItemDescriptor => {
+  ): PanelSelectionMenuItem => {
     const { name, title, description, icon = 'empty' } = visTypeAlias;
 
     return {
+      id: name,
       name: title,
       icon,
-      onClick: createNewVisType(visTypeAlias),
+      onClick: augmentedCreateNewVisType(visTypeAlias, onClickCb),
       'data-test-subj': `visType-${name}`,
-      toolTipContent: description,
+      description,
     };
   };
 
-  const aggsPanelTitle = i18n.translate('dashboard.editorMenu.aggBasedGroupTitle', {
-    defaultMessage: 'Aggregation based',
-  });
+  const getEditorMenuPanels = (closeFlyout: () => void): GroupedAddPanelActions[] => {
+    const getEmbeddableFactoryMenuItem = getEmbeddableFactoryMenuItemProvider(api, closeFlyout);
 
-  const getEditorMenuPanels = (closePopover: () => void): EuiContextMenuPanelDescriptor[] => {
-    const getEmbeddableFactoryMenuItem = getEmbeddableFactoryMenuItemProvider(api, closePopover);
-
-    const [ungroupedAddPanelActions, groupedAddPanelAction] = getAddPanelActionMenuItems(
+    const groupedAddPanelAction = getAddPanelActionMenuItemsGroup(
       api,
       addPanelActions,
-      closePopover
+      closeFlyout
     );
 
-    const [initialPanelGroups, additionalPanels] = mergeGroupedItemsProvider(
-      getEmbeddableFactoryMenuItem
-    )(factoryGroupMap, groupedAddPanelAction);
-
-    const initialPanelItems = [
-      ...visTypeAliases.map(getVisTypeAliasMenuItem),
-      ...ungroupedAddPanelActions,
-      ...toolVisTypes.map(getVisTypeMenuItem),
-      ...ungroupedFactories.map(getEmbeddableFactoryMenuItem),
-      ...initialPanelGroups,
-      ...promotedVisTypes.map(getVisTypeMenuItem),
-    ];
-    if (aggsBasedVisTypes.length > 0) {
-      initialPanelItems.push({
-        name: aggsPanelTitle,
-        icon: 'visualizeApp',
-        panel: aggBasedPanelID,
-        'data-test-subj': `dashboardEditorAggBasedMenuItem`,
-      });
-    }
+    const initialPanelGroups = mergeGroupedItemsProvider(getEmbeddableFactoryMenuItem)(
+      factoryGroupMap,
+      groupedAddPanelAction
+    );
 
-    return [
-      {
-        id: 0,
-        items: initialPanelItems,
-      },
-      {
-        id: aggBasedPanelID,
-        title: aggsPanelTitle,
-        items: aggsBasedVisTypes.map(getVisTypeMenuItem),
-      },
-      ...additionalPanels,
-    ];
+    // enhance panel groups
+    return sortGroupPanelsByPlacementPriority(initialPanelGroups).map((panelGroup) => {
+      switch (panelGroup.id) {
+        case 'visualizations': {
+          return {
+            ...panelGroup,
+            items: (panelGroup.items ?? []).concat(
+              promotedVisTypes.map(getVisTypeMenuItem.bind(null, closeFlyout)),
+              // TODO: actually add grouping to vis type alias so we wouldn't randomly display an unintended item
```

Comment:
Is this TODO something that still needs to be done in this PR? or is this work to be done as follow up?

Reply:
This would be a follow up item.